doc hackathon additions to cli docs and opening w/ a specific database (#17601)
visit-dav/visit: src/doc/python_scripting/functions.rst

  SetPrecisionType(typeAsString)

> typeAsInt : integer

Author: This scares me, since someone thought it should be a double. But everything I can find suggests that it is in fact an integer.

Reviewer: I looked at the code and tested it as well. It takes the following args:

0, "Float"
1, "Native"
2, "Double"

The capitalization matters. The default is "Native".

  SetPrecisionType(typeAsString)

typeAsInt : integer

>     Precision type specified as an integer. 0 = float 1 = native 2 = double.

Reviewer: There is a typo a few lines down on line 10136. "pecision" should be "precision".

WriteScript
-----------

**Synopsis:**

::

>   WriteScript(f)

Author: Where can I find the return type for this function? I feel like I looked everywhere.

Author: It returns Nonetype

f : file
    The python file object that will be written to.

> 

Author: Where can I find the return type for this function? I feel like I looked everywhere.

Reviewer 2: I think its in `src/visitpy/visit_utils/src/builtin/writescript.py`.

Ultimately, the VisIt python interface is a mixture of C/C++ Extension modules to Python, pure python functions, and other python modules (e.g. `numpy`, `pyside`, etc.). `WriteScript` is a pure python function.

WriteScript
-----------

**Synopsis:**

::

  WriteScript(f)

f : file
    The python file object that will be written to.

**Description:**

>     WriteScript is meant to be able to write out the current state of VisIt to a Python script that can be used later to reproduce a visualization. 

Reviewer 2: ```suggestion
    ``WriteScript()`` saves the current state of VisIt as a Python script that can be used later to reproduce a visualization. 
```

>     This is like using a session file but you can take the output of WriteScript and further customize it.

Reviewer 2: ```suggestion
    This is like saving a session file. But, the output of WriteScript can be further customized.
```

**Example:**

::

>   #% visit -cli

Reviewer 2: WriteScript was recently modified to accept either an open file handle (as it was originally coded) or just the name of a file. So, I am proposing changes here that document both approaches...

```suggestion
  #% visit -cli
  OpenDatabase("foo.silo")
  AddPlot("Pseudocolor","dx")
  DrawPlots()
  ChangeActivePlotsVar("dy")
  WriteScript("plot_dx_and_dy.py")

or

::

    #% visit -cli  
```

>   f = open("script.py", "wt")

Reviewer 2: ```suggestion
  OpenDatabase("foo.silo")
  AddPlot("Pseudocolor","dx")
  DrawPlots()
  ChangeActivePlotsVar("dy")
  f = open("plot_dx_and_dy.py", "wt")
```

  WriteScript(f)
  f.close()

The resulting script will contain commands to set up plots in any visualization window that contained plots when WriteScript was called. 

> You may want to edit the resulting script file since it may be verbose, depending on how many plots you have or the complexity of the data file. For example, you might want to remove the code related to setting a plot's SIL restriction.

Reviewer 2: ```suggestion
The resulting script may be more verbose than necessary. Users may find it useful to delete portions of the script that are not needed.  This will depend on how many plots there are or the complexity of the data. For example, it might useful to remove code related to setting a plot's SIL restriction.
```

> Once you have edited the script to your satisfaction, you can replay it in VisIt like this:

Reviewer 2: ```suggestion
Once the script is edited to satisfaction, it can be replayed it in VisIt like this:
```

::

  visit -cli -s script.py

> Or, you can even use the script with VisIt's movie scripts as a means to set up your visualization: 

Reviewer 2: ```suggestion
Or, the script can even be used with VisIt's movie making scripts as a basis to set up the initial visualization: 
```
